tests/test_physics/test_ansatz.py

```diff
@@ -155,3 +154,102 @@ def test_PolyExpAnsatz_simplify_v2(Abc):
     assert len(ansatz.A) == 1
     assert len(ansatz.b) == 1
     assert np.allclose(ansatz.c, 2 * c)
+
+
+class TestArrayAnsatz:
+    r"""Tests all algebra related to ArrayAnsatz."""
+
+    def test_ArrayAnsatz_init_(self):
+        r"""Tests that an ArrayAnstaz can be initialized."""
+        array = np.random.random((2, 4, 5))
+        aa = ArrayAnsatz(array=array)
+        assert isinstance(aa, ArrayAnsatz)
+        assert np.allclose(aa.array, array)
+
+    def test_ArrayAnsatz_neg(self):
+        r"""Negates the array inside ArrayAnsatz."""
+        array = np.random.random((2, 4, 5))
+        aa = ArrayAnsatz(array=array)
+        minusaa = -aa
+        assert isinstance(minusaa, ArrayAnsatz)
+        assert np.allclose(minusaa.array, -array)
+
+    def test_ArrayAnsatz_equal(self):
+        r"""Tests the equation of two ArrayAnsatzs."""
+        array = np.random.random((2, 4, 5))
+        aa1 = ArrayAnsatz(array=array)
+        aa2 = ArrayAnsatz(array=array)
+        assert aa1 == aa2
+
+    def test_ArrayAnsatz_addition(self):
+        r"""Tests the correctness of adding two ArrayAnsatzs."""
+        array = np.random.random((2, 4, 5))
+        array2 = np.random.random((4, 4, 5))
+        aa1 = ArrayAnsatz(array=array)
+        aa2 = ArrayAnsatz(array=array2)
+        aa1_add_aa2 = aa1 + aa2
+        assert isinstance(aa1_add_aa2, ArrayAnsatz)
+        assert aa1_add_aa2.array.shape == (8, 4, 5)
+
+    def test_ArrayAnsatz_and(self):
+        r"""Tests the correctness of adding two ArrayAnsatzs."""
+        array = np.random.random((2, 4, 5))
+        array2 = np.random.random((7, 4, 5))
+        aa1 = ArrayAnsatz(array=array)
+        aa2 = ArrayAnsatz(array=array2)
+        aa1_and_aa2 = aa1 & aa2
+        assert isinstance(aa1_and_aa2, ArrayAnsatz)
+        assert aa1_and_aa2.array.shape == (14, 4, 5, 4, 5)
```

Comment:
If the arrays were smaller, could you test the actual values? And same for the above. Because here you are only testing the shapes

Reply:
Maybe you could try with non-random arrays, like `array = np.arange(8).reshape(2,2,2)` and `array1=array.reshape(1, 2, 4)`